File: 04/collatz.py
def collatz(n):
    if (n % 2 == 0):
        n //= 2
    else:
        n = 3*n + 1
    return n

def collatzSequence():
    num = int(input("Enter Number: "))
    print(num)
    while num != 1:
        num = collatz(num)
        print(num)
        
        
collatzSequence()

File: 13-python2c++/collatz.cpp
#include <iostream>

int collatz(int n){
  if(n % 2 == 0){
    n /= 2;
  }
  else{
    n = 3 * n + 1;
  }
  return n;
}

int main(){
  int num;
  std::cout << "Collatz. Enter number : ";
  std::cin >> num;

  while(num > 1){
    num = collatz(num);
    std::cout << num << "\n";
  }
  std::cout << "done\n" ;
  return 0;
}